### ISSUE TITLE
improve unreachable rule - hoisted functions are marked incorrectly.

### ISSUE THREAD

**Issue by jeme** [palantir/tslint#100]
In relation to #31 and #83.

It doesn't seem like the fix for those also fixes false-negatives for hoisted functions.
The code below could probably be used as a bases for the test-case covering this.

``` javascript
function hoistingBug() { 
  return hoisted();

  function hoisted() {
    return 'function declaration \'hoisted\' and the function body gets hoisted, meaning this code is not unreachable.';
  }
}
```
